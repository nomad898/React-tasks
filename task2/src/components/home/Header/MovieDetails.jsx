import PropTypes from 'prop-types';
import {
    Text,
    RedText
} from '@components/global';
import {
    MovieDetails as Wrapper,
    MovieDetailsImageCol,
    MovieDetailsInfoCol,
    MovieDetailsInfoRow,
    MovieDetailsTitleRow,
    MovieDetailsImage,
    MovieDetailsTitle,
    MovieDetailsRating,
    MovieDetailsDurationText,
    MovieDetailsDescription,
} from './MovieDetails.styles';

const MovieDetails = (
    {
        movie
    }) => (
    <Wrapper>
        <MovieDetailsImageCol>
            <MovieDetailsImage src={movie.poster} />
        </MovieDetailsImageCol>
        <MovieDetailsInfoCol>
            <MovieDetailsTitleRow>
                <MovieDetailsTitle>
                    {movie.title}
                </MovieDetailsTitle>
                <MovieDetailsRating>
                    <span>
                        {movie.rating}
                    </span>
                </MovieDetailsRating>
            </MovieDetailsTitleRow>
            <MovieDetailsInfoRow>
                <Text>
                    {movie.genres}
                </Text>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <RedText>
                    {movie.releaseDate.getFullYear()}
                </RedText>
                <MovieDetailsDurationText>
                    {movie.duration} min
                </MovieDetailsDurationText>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <MovieDetailsDescription>
                    {movie.description}
                </MovieDetailsDescription>
            </MovieDetailsInfoRow>
        </MovieDetailsInfoCol>
    </Wrapper>
);

MovieDetails.propTypes = {
    movie: PropTypes.shape({
        title: PropTypes.string,
        releaseDate: PropTypes.instanceOf(Date),
        genres: PropTypes.string,
        poster: PropTypes.string,
        rating: PropTypes.number.isRequired,
        duration: PropTypes.number.isRequired,
        description: PropTypes.string.isRequired
    })
};

export { MovieDetails }