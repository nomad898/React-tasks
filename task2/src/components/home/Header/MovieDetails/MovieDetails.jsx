import PropTypes from 'prop-types';
import {
    Text,
    RedText
} from '@components/global';
import {
    MovieDetails as Wrapper,
    MovieDetailsImageCol,
    MovieDetailsInfoCol,
    MovieDetailsInfoRow,
    MovieDetailsTitleRow,
    MovieDetailsImage,
    MovieDetailsTitle,
    MovieDetailsRating,
    MovieDetailsDurationText,
    MovieDetailsDescription,
} from './MovieDetails.styles';

const MovieDetails = (
    {
        movie
    }) => (
    <Wrapper>
        <MovieDetailsImageCol>
            <MovieDetailsImage src={movie.posterPath} />
        </MovieDetailsImageCol>
        <MovieDetailsInfoCol>
            <MovieDetailsTitleRow>
                <MovieDetailsTitle>
                    {movie.title}
                </MovieDetailsTitle>
                <MovieDetailsRating>
                    <span>
                        {movie.voteAverage}
                    </span>
                </MovieDetailsRating>
            </MovieDetailsTitleRow>
            <MovieDetailsInfoRow>
                <Text>
                    {movie.genres.join(', ')}
                </Text>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <RedText>
                    {movie.releaseDate.getFullYear()}
                </RedText>
                <MovieDetailsDurationText>
                    {movie.runtime} min
                </MovieDetailsDurationText>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <MovieDetailsDescription>
                    {movie.overview}
                </MovieDetailsDescription>
            </MovieDetailsInfoRow>
        </MovieDetailsInfoCol>
    </Wrapper>
);

MovieDetails.propTypes = {
    movie: PropTypes.shape({
        title: PropTypes.string,
        releaseDate: PropTypes.instanceOf(Date),
        genres: PropTypes.arrayOf(PropTypes.string),
        posterPath: PropTypes.string,
        voteAverage: PropTypes.number.isRequired,
        runtime: PropTypes.number.isRequired,
        overview: PropTypes.string.isRequired
    }).isRequired
};

export { MovieDetails }