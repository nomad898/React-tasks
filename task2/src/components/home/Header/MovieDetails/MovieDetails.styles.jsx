import styled from 'styled-components';
import color from '@styles/modules/_colors.module.scss';
import { Text, RedText } from '@components/global';

const MovieDetails = styled.div`
    display: flex;
    padding: 1em 1.5em;
`;

const MovieDetailsImageCol = styled.div`
    width: 100%;
`;

const MovieDetailsInfoCol = styled.div`
    margin: 0.6em 1.6em;
`;

const MovieDetailsImage = styled.img`
    width: 100%;
`;


const MovieDetailsInfoRow = styled.div`
    margin: 1em 0.6em;
`;

const MovieDetailsTitleRow = styled(MovieDetailsInfoRow)`
    display: flex;
    align-items: center;
`;

const MovieDetailsTitle = styled(Text)`
    font-size: 3em;
`;

const MovieDetailsRating = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    border: 2px solid ${color.white};
    color: ${color.green};
    height: 2em;
    width: 2em;
    margin: 0 1em;
`;

const MovieDetailsDurationText = styled(RedText)`
    margin: 0 2em;
`;

const MovieDetailsDescription = styled.p`
    color: ${color.white}
`;

export {
    MovieDetails,
    MovieDetailsImageCol,
    MovieDetailsInfoCol,
    MovieDetailsInfoRow,
    MovieDetailsTitleRow,
    MovieDetailsImage,
    MovieDetailsTitle,
    MovieDetailsRating,
    MovieDetailsDurationText,
    MovieDetailsDescription,
};