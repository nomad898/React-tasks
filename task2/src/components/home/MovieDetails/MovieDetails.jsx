import PropTypes from 'prop-types';
import {
    Text,
    RedText
} from '@components/global';
import {
    MovieDetails as Wrapper,
    MovieDetailsImageCol,
    MovieDetailsInfoCol,
    MovieDetailsInfoRow,
    MovieDetailsTitleRow,
    MovieDetailsImage,
    MovieDetailsTitle,
    MovieDetailsRating,
    MovieDetailsDurationText,
    MovieDetailsDescription,
} from './MovieDetails.styles';

const MovieDetails = (props) => (
    <Wrapper>
        <MovieDetailsImageCol>
            <MovieDetailsImage src={props.movie.poster} />
        </MovieDetailsImageCol>
        <MovieDetailsInfoCol>
            <MovieDetailsTitleRow>
                <MovieDetailsTitle>
                    {props.movie.title}
                </MovieDetailsTitle>
                <MovieDetailsRating>
                    <span>
                        {props.movie.rating}
                    </span>
                </MovieDetailsRating>
            </MovieDetailsTitleRow>
            <MovieDetailsInfoRow>
                <Text>
                    {props.movie.genres}
                </Text>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <RedText>
                    {props.movie.releaseDate.getFullYear()}
                </RedText>
                <MovieDetailsDurationText>
                    {props.movie.duration} min
                </MovieDetailsDurationText>
            </MovieDetailsInfoRow>
            <MovieDetailsInfoRow>
                <MovieDetailsDescription>
                    {props.movie.description}
                </MovieDetailsDescription>
            </MovieDetailsInfoRow>
        </MovieDetailsInfoCol>
    </Wrapper>
);

MovieDetails.propTypes = {
    movie: PropTypes.shape({
        title: PropTypes.string,
        releaseDate: PropTypes.instanceOf(Date),
        genres: PropTypes.string,
        poster: PropTypes.string,
        rating: PropTypes.number.isRequired,
        duration: PropTypes.number.isRequired,
        description: PropTypes.string.isRequired
    })
};

export { MovieDetails }