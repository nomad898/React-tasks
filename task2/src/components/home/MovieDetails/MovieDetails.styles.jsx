import styled from 'styled-components';
import { Text, RedText } from '@components/global';
import { Colors } from '@utils';

const MovieDetails = styled.div`
    display: flex;
    padding: 1em 1.5em;
`;

const MovieDetailsImageCol = styled.div`
    width: 100%;
`;

const MovieDetailsInfoCol = styled.div`
    margin: 0.6em 1.6em;
`;

const MovieDetailsImage = styled.img`
    width: 100%;
`;


const MovieDetailsInfoRow = styled.div`
    margin: 1em 0.6em;
`;

const MovieDetailsTitleRow = styled(MovieDetailsInfoRow)`
    display: flex;
    align-items: center;
`;

const MovieDetailsTitle = styled(Text)`
    font-size: 3em;
`;

const MovieDetailsRating = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    border: 2px solid ${Colors.WHITE};
    color: ${Colors.GREEN};
    height: 2em;
    width: 2em;
    margin: 0 1em;
`;

const MovieDetailsDurationText = styled(RedText)`
    margin: 0 2em;
`;

const MovieDetailsDescription = styled.p`
    color: ${Colors.WHITE}
`;

export {
    MovieDetails,
    MovieDetailsImageCol,
    MovieDetailsInfoCol,
    MovieDetailsInfoRow,
    MovieDetailsTitleRow,
    MovieDetailsImage,
    MovieDetailsTitle,
    MovieDetailsRating,
    MovieDetailsDurationText,
    MovieDetailsDescription,
};