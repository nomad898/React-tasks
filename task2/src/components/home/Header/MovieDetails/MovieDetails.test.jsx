import { render, cleanup } from '@testing-library/react';
import { MovieDetails } from './MovieDetails';
import { Movie } from '@models';

afterEach(cleanup);

const movie = new Movie(
    "Jumanji: Welcome to the Jungle",
    "The game has evolved.",
    6.5,
    3330,
    new Date("2017-12-09"),
    "https://image.tmdb.org/t/p/w500/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
    "The tables are turned as four teenagers are sucked into Jumanji's",
    90000000,
    928901353,
    [
        "Action",
        "Adventure",
        "Comedy",
        "Family"
    ],
    119,
    353486
);

it('should take a snapshot', () => {
    const { asFragment } = render(<MovieDetails movie={movie}/>);

    expect(asFragment(<MovieDetails movie={movie}/>)).toMatchSnapshot();
})