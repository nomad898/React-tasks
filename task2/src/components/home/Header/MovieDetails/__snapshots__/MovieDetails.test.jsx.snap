// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fnVZcZ gaVJhY"
  >
    <div
      class="sc-fFSPTT PQcM"
    >
      <img
        class="sc-iemWCZ cVmCtg"
        src="https://image.tmdb.org/t/p/w500/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg"
      />
    </div>
    <div
      class="sc-bkbkJK bTyyDD"
    >
      <div
        class="sc-dIvrsQ sc-hHEiqL dDscFh kHSXvr"
      >
        <span
          class="sc-jSFjdj sc-dlMDgC bFcnby ceiYFo"
        >
          Jumanji: Welcome to the Jungle
        </span>
        <div
          class="sc-kfYoZR fVZbyK"
        >
          <span>
            6.5
          </span>
        </div>
      </div>
      <div
        class="sc-dIvrsQ dDscFh"
      >
        <span
          class="sc-jSFjdj bFcnby"
        >
          Action, Adventure, Comedy, Family
        </span>
      </div>
      <div
        class="sc-dIvrsQ dDscFh"
      >
        <span
          class="sc-iCoGMd jqNton"
        >
          2017
        </span>
        <span
          class="sc-iCoGMd sc-fKgJPI jqNton iZySEA"
        >
          119 min
        </span>
      </div>
      <div
        class="sc-dIvrsQ dDscFh"
      >
        <p
          class="sc-bCwfaz eBnsE"
        >
          The tables are turned as four teenagers are sucked into Jumanji's
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
